src/controllers/reservationController.js

```diff
@@ -0,0 +1,76 @@
+import mongoose from 'mongoose';
+import { Reservation } from '../models/Reservation.js';
+import { RealEstateErrors } from '../utils/ErrorHandler.js';
+
+export const createReservation = async (req, res, next) => {
+  try {
+    const reservation = new Reservation(req.body);
+    await reservation.save();
+
+    const { _id, listingId, startDate, endDate } = reservation;
+    res.status(200).send({ id: _id, listingId, startDate, endDate });
+  } catch {
+    next(new RealEstateErrors());
+  }
+};
+
+export const getAllReservations = async (req, res, next) => {
+  try {
+    const reservations = await Reservation.find();
+    const filteredReservations = reservations.map((reservation) => ({
+      id: reservation._id,
+      listingId: reservation.listingId,
+      startDate: reservation.startDate,
+      endDate: reservation.endDate,
+    }));
+
+    res.status(200).send(filteredReservations);
```

Comment:
Try to include in the response meta information like `totalCount`.
For example:
```json
{
  "totalCount": 200,
  "data": [
    {
      "id": ...
    }
  ]
}
```

Reply:
Also, you can apply this to other endpoints as well. Get endpoints that is.